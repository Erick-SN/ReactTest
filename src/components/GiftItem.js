const GiftItem = ({ id, title, url }) => {
  return (
    <>
      <div className="card">
        <img src={url} alt="animated" />
        <p>{title}</p>
      </div>
    </>
  );
};

export default GiftItem;
