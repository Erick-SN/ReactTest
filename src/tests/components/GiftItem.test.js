import { shallow } from 'enzyme';
import GiftItem from '../../components/GiftItem';

describe('Pruebas en <GiftItem/>', () => {
  let wrapper;
  const title = 'title';
  const url = 'http://';
  beforeEach(() => {
    wrapper = shallow(<GiftItem title={title} url={url} />);
  });
  test('Renderizado de componente', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('Asegurando los elemtos para mostrar la imagen', () => {
    const p = wrapper.find('p');
    const img = wrapper.find('img');
    expect(p.text().trim()).toBe(title);
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });
});
