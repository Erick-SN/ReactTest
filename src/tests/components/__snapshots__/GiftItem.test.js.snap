// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftItem/> Renderizado de componente 1`] = `
<Fragment>
  <div
    className="card animate__animated animate__fadeIn"
  >
    <img
      alt="title"
      src="http://"
    />
    <p>
      title
    </p>
  </div>
</Fragment>
`;
